SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -317,6 +345,25 @@ void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
 				Actor->OnAuthorityLost();
 			}
 		}
+
+		// Subobject Delegation
+		TPair<Worker_EntityId_Key, Worker_ComponentId> EntityComponentPair = MakeTuple(Op.entity_id, Op.component_id);
+		if (TSharedRef<FPendingSubobjectAttachment>* PendingSubobjectAttachmentPtr = PendingEntitySubobjectDelegations.Find(EntityComponentPair))
+		{
+			FPendingSubobjectAttachment& PendingSubobjectAttachment = PendingSubobjectAttachmentPtr->Get();
+
+			PendingSubobjectAttachment.PendingAuthorityDelegations.Remove(Op.component_id);
+
+			if (PendingSubobjectAttachment.PendingAuthorityDelegations.Num() == 0)
+			{
+				if (UObject* Object = PendingSubobjectAttachment.Subobject.Get())
+				{
+					Sender->SendAddComponent(PendingSubobjectAttachment.Channel, Object, *PendingSubobjectAttachment.Info);
+				}
```

Comment:
Should we warn if subobject is destroyed before we could add that component?

Reply:
Hmm, dunno. I think that's potentially valid? If you try to add a component then immediately delete the entity, that is valid game behavior even though there might not be time to actually attach the component.